mernarezk/assignment4/CountIslands.java

```diff
@@ -0,0 +1,36 @@
+package Assignment4;
+
+public class CountIslands {
+	/**
+	 * Iterate over all valid start points, which are set to true meaning they're land
+	 * and try to recursively explore from that starting point setting all the land cells to
+	 * false so that we do not visit a cell twice and we count a land everytime we have a valid
+	 * starting point.
+	 * @param grid that needs to be explored.
+	 * @param m number of rows.
+	 * @param n number of columns.
+	 * @return number of islands in the grid.
+	 */
+	public int countIslands(boolean grid[][], int m, int n) {

```

Comment:
Do we really need the row and column count as input parameters? I think we can deduce the size from the grid array size.

Reply:
I was just told that it was in the description of the problem to use these as parameters. Feel free to either keep them or remove them, both ways are okay. :)

Reply:
Ah yes I added it because of description really.